### PR TITLE
Added provider management pages in.

### PR DESCRIPTION
**Pull request by minglis**
Depends on: https://github.com/alphagov/notifications-api/pull/295

Adds platform admin pages to mange priority of the providers.

- list all providers
- update the priority

### CODE REVIEW

File: app/main/views/providers.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@main.route("/provider/<provider_id>", methods=['GET', 'POST'])
+@login_required
+@user_has_permissions(admin_override=True)
+def view_provider(provider_id):
+
+    provider = provider_client.get_provider_by_id(provider_id)['provider_details']
+
+    form = ProviderForm(active=provider['active'], priority=provider['priority'])
+
+    print(form)
```
Comment (mannickutd) [inline app/main/views/providers.py]: Please remove print statements.

File: tests/app/main/views/test_providers.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        {
+            'active': True,
+            'priority': 2,
+            'display_name': 'second_email_provider',
+            'identifier': 'second_email',
+            'id': '0bd529cd-a0fd-43e5-80ee-b95ef6b0d51b',
+            'notification_type': 'email'
+        }
+    ]
+}
```
Comment (mannickutd) [inline tests/app/main/views/test_providers.py]: These would make great functions in the tests/init.py
def provider_json(<include arguments>)

Bonus for having these as test fixtures.

File: app/main/views/providers.py (hunk 1 above)
Comment (minglis) [inline app/main/views/providers.py]: doh

Comment (minglis) [inline app/main/views/providers.py]: fixed

File: tests/app/main/views/test_providers.py (hunk 2 above)
Comment (minglis) [inline tests/app/main/views/test_providers.py]: Thinking of leaving these as these are only relevant to these tests?

**Pull request merged**